### PR TITLE
ref(emotion11): Improve some legacy Form types

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/components/forms/formField.tsx [hunk 1]
```diff
@@ -40,13 +43,13 @@ export default class FormField<
...
 
-  static defaultProps = {
+  static defaultProps: DefaultProps = {
     hideErrorMessage: false,
     disabled: false,
     required: false,
   };
 
-  constructor(props: Props, context: Context) {
+  constructor(props: Props, context?: any) {
```
Comment (EvanPurkhiser) [inline static/app/components/forms/formField.tsx:52]: emotion 11 is NOT happy about typing this legacy prop

File: static/app/components/forms/selectField.tsx [hunk 2]
```diff
@@ -8,22 +8,11 @@ import FormField from './formField';
 import SelectControl, {ControlProps} from './selectControl';
 
 type SelectProps = Omit<ControlProps, 'onChange' | 'name'>;
-type FormProps = Omit<FormField['props'], 'onChange' | 'name'>;
+type FormProps = FormField['props'];
 
-type Props = {
-  name: string;
-  onChange: FormField['props']['onChange'];
-  deprecatedSelectControl?: boolean;
```
Comment (EvanPurkhiser) [inline static/app/components/forms/selectField.tsx:16]: Simplified

File: static/app/components/forms/selectField.tsx [hunk 3]
```diff
@@ -8,22 +8,11 @@ import FormField from './formField';
...
-} & FormProps &
-  SelectProps;
+type Props = FormProps & SelectProps;
 
 export default class SelectField extends FormField<Props> {
-  static defaultProps = {
```
Comment (EvanPurkhiser) [inline static/app/components/forms/selectField.tsx:21]: Nope, these need to come back, oops

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Pull request merged**